core/domain/rights_manager.py

```diff
@@ -623,6 +658,41 @@ def check_can_edit_activity(user, activity_rights):
     return False
 
 
+def check_can_translate_activity(user, activity_rights):
+    """Checks whether the user can translate given activity.
+
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        activity_rights: ActivityRights or None. Rights object for the given
+            activity.
+
+    Returns:
+        bool. Whether the given user can translate this activity.
+    """
+    if activity_rights is None:
+        return False
+
+    if role_services.ACTION_EDIT_OWNED_ACTIVITY not in user.actions:
```

Comment:
The flow here seems a bit off to me. It assumes that it's impossible for a user to have ACTION_EDIT_ANY_ACTIVITY but not ACTION_EDIT_OWNED_ACTIVITY. While currently true, this is an implicit assumption.

So perhaps you should check the last couple of things (any activity, any public activity) first, so that you don't erroneously exit early.

Reply:
Yes, I think so.